### PR TITLE
Update Company Command and Delete Command

### PR DESCRIPTION
**Pull request by amoonguss1**
As per title, resolves issues #159

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -216,7 +216,7 @@ This section describes some noteworthy details on how certain features are imple
 
 ### Add Command
 
-The `add` command is used to create a new role in the app and set the necessary fields for that role,
+The `add` command is used to create a new role in TechTrack and set the necessary fields for that role,
```
Comment (dsja612) [inline docs/DeveloperGuide.md:219]: nice

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -328,32 +350,51 @@ will sort both views.
...
 user to filter the job list by company which shows all roles pertaining to a certain company.
 
-The feature uses operations in the `Model` interface as `Model#updateFilteredRoleList()`.
-
-Given below is an example usage of how CompanyCommand is being used in the following steps.
+Given below is an example usage of how CompanyCommand and its dependencies is being used in the following steps.
 
 1. The user launches the application for the first time. The `RoleBook` will be initialized with the
-   current role book.
+   current role book from the data files.
```
Comment (dsja612) [inline docs/DeveloperGuide.md:356]: Maybe "The user launches TechTrack, and initializes its role book from preexisting data files" would be better

cuz there is no data file on a fresh installation

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -559,7 +600,7 @@ Use case ends.
 
 **Extensions**
 
-* 3a. Job exists.
+* 3a. Job is valid.
```
Comment (dsja612) [inline docs/DeveloperGuide.md:603]: Later we need to change `job` to `role` probably after we all done with the important stuff

**Review by dsja612 — CHANGES_REQUESTED** [review]
change 1 line can liao aft that lgtm

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (amoonguss1) [reply docs/DeveloperGuide.md:356]: 🤓

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (amoonguss1) [reply docs/DeveloperGuide.md:602]: 🤓

**Review by dsja612 — APPROVED** [review]
lgtm

**Pull request merged**